README.md

```diff
@@ -1 +1,293 @@
+migrations leaves the user with no way to run the migrations because there is no CLI in the browser. In such
+a case, you should provide a way to trigger migrations manually.**
+
+### Writing migration
+
+Migrations are one of those things that can be extremely painful on users. At the end of the day, we want users never to have to think about it. The process should be:
+
+- SAFE. No data lost. Ever.
+- Revertible. Tools must implement forward and backward (if possible) migrations.
+- Tests. Migrations have to be well tested.
+- To Spec. The tools must conform to the spec.
+
+If your migration has several parts, it should be fail-proof enough that if one part of migration fails the previous changes
+are reverted before propagating the error. If possible then the outcome should be consistent repo so it migration could
+be run again. 
+
+#### Architecture of migrations
+
+All migrations are placed in the `/migrations` folder. Each folder there represents one migration that follows the migration
+API.
+
+All migrations are collected in `/migrations/index.js`, which should not be edited manually. It is regenerated on
+every run of `jsipfs-migrations add` (manual changes should follow the same style of modifications). 
+**The order of migrations is important and migrations must be sorted in ascending order**.
+
+Each migration must follow this API. It must export an object in its `index.js` that has following properties:
+
+ * `version` (int) - Number that represents the version which the repo will migrate to (eg. `migration-8` will move the repo to version 8).
+ * `description` (string) - Brief description of what the migrations does.
+ * `migrate` (function) - Function that performs the migration (see signature of this function below)
+ * `revert` (function) - If defined then this function will revert the migration to the previous version. Otherwise it is assumed that it is not possible to revert this migration.
+
+##### `migrate(repoPath, isBrowser)`
+
+_Do not confuse this function with the `require('ipfs-repo-migrations').migrate()` function that drives the whole migration process!_
+
+Arguments:
+ * `repoPath` (string) - absolute path to the root of the repo
+ * `options` (object, optional) - object containing `IPFSRepo` options, that should be used to construct a datastore instance.
+ * `isBrowser` (bool) - indicates if the migration is run in a browser environment (as opposed to NodeJS)
+ 
+##### `revert(repoPath, isBrowser)`
+
+_Do not confuse this function with the `require('ipfs-repo-migrations').revert()` function that drives the whole backward migration process!_
+
+Arguments:
+ * `repoPath` (string) - path to the root of the repo
+ * `options` (object, optional) - object containing `IPFSRepo` options, that should be used to construct the datastore instance.
+ * `isBrowser` (bool) - indicates if the migration is run in a browser environment (as opposed to NodeJS)
+
+#### Browser vs. NodeJS environments
```

Comment:
This could be done automatically, no? All it should care about is the path

Reply:
Yes, as mentioned lower in most of the cases the difference is handled by datastore implementation. But there might be a need for this in some more complicated migrations.